modules/compute/gke-node-pool/main.tf

```diff
@@ -283,6 +283,22 @@ resource "google_container_node_pool" "node_pool" {
       )
       error_message = "Shared extended reservations are not supported by GKE."
     }
+    precondition {
+      condition     = contains(["SURGE"], var.upgrade_settings.strategy)
+      error_message = "Only SURGE strategy is supported"
+    }
+    precondition {
+      condition     = var.upgrade_settings.max_unavailable >= 0
+      error_message = "max_unavailable should be set to 0 or greater"
+    }
+    precondition {
+      condition     = var.upgrade_settings.max_surge >= 0
+      error_message = "max_surge should be set to 0 or greater"
+    }
+    precondition {
+      condition     = (var.upgrade_settings.max_unavailable > 0 || var.upgrade_settings.max_surge > 0)
+      error_message = "At least one of max_unavailable or max_surge must greater than 0"
+    }
```

Comment:
Isn't it possible that these parameters 
```
    max_surge       = optional(number),
    max_unavailable = optional(number)
```
can be null? Is that case handled gracefully?

Reply:
This seems to be poorly documented. Will add test results for `{strategy:"SURGE"}`.

Reply:
Ran multiple tests. There seems to be odd behaviors observed when one of the parameters get set to null. Based on offline discussions. If the values are not specified, we would use default parameters. Description would be updated to reflect the same. Some of the edge cases have been handled in the [kubernetes code base](https://github.com/kubernetes/kubernetes/blob/a499facee693a1a83daadb82d88f7b51d324ffc5/pkg/controller/daemon/update.go#L612-L615)